### PR TITLE
Add SQG vertical structure to eddy diffusivities

### PR DESCRIPTION
**Pull request by Wendazhang33**
This commit added an surface quasigeostrophy (SQG) vertical structure (sqg_struct) in Varmix (MOM_lateral_mixing_coeffs) to provide a vertical profile for eddy diffusivities. The SQG vertical structure and its parameterization is given by Zhang et al. (2024): https://journals.ametsoc.org/view/journals/phoc/54/6/JPO-D-23-0203.1.xml. 
Also, the vertical structure of all eddy diffusivities/viscosities, including khth, khtr, kh, and kdgl90, are reorganized in Varmix. Two vertical structures are optional for each diffusivity: the equivalent barotropic (ebt_struct) and surface quasigeostrophy (sqg_struct) structures. For example, if one wants to use the SQG structure for the thickness diffusivity Khth, they can set KHTH_USE_SQG_STRUCT=True. 
Detailed changes are summarized below:
- KHTH_USE_EBT_STRUCT, KHTR_USE_EBT_STRUCT, KDGL90_USE_EBT_STRUCT and BS_EBT_POWER parameters, which already existed, still control whether to use the EBT structure for khth, khtr, kdgl90, and kh, respectively
- Added KHTH_USE_SQG_STRUCT, KHTR_USE_SQG_STRUCT, KDGL90_USE_SQG_STRUCT and BS_USE_SQG parameters to control whether to use the SQG structure for khth, khtr, kdgl90, and kh, respectively
- Added function calc_sqg_struct in MOM_lateral_mixing_coeffs to compute sqg_struct
- The sqg_struct can be tuned by the exponent sqg_expo, default is 1.0, suggested range is 0.5-3.0
- If SQG_USE_MEKE=True, the eddy length scale from MEKE will be used to compute sqg_struct
- The eddy length scale, Le, will be added into MEKE and registered as a restart variable, MEKE_LE, if SQG_USE_MEKE=True
- MEKE and dt are added in Varmix to compute sqg_struct

### CODE REVIEW

**Comment by adcroft** [comment]
This look good to me. The only question I had was whether `MEKE%Le` is the right length scale to pass in the code, given `Le` can be configured to be pretty much anything. @Wendazhang33 tells me that it only makes sense when MEKE is using the minimum of the Rhines scale or grid-spacing. I'll wait for @Hallberg-NOAA's opinion.

File: src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90 [hunk 1]
```diff
@@ -470,6 +535,74 @@ subroutine calc_resoln_function(h, tv, G, GV, US, CS)
...
+
+  call find_eta(h, tv, G, GV, US, e, halo_size=2)
+  call calc_isoneutral_slopes(G, GV, US, h, e, tv, dt*CS%kappa_smooth, CS%use_stanley_iso, &
+                                  CS%slope_x, CS%slope_y, N2_u=N2_u, N2_v=N2_v,dzu=dzu, dzv=dzv, &
+                                  dzSxN=dzSxN, dzSyN=dzSyN, halo=1)
+
+  do j=js,je ; do i=is,ie
+    CS%sqg_struct(i,j,1) = 1.0
+  enddo ; enddo
+  if (allocated(MEKE%Le)) then
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90:577]: Given the expense of the calculations, there should be a separate logical branch here to handle the case where `CS%sqg_expo == 0.0`, as this is both the default value and is much simpler to calcluate.

File: src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90 [hunk 2]
```diff
@@ -470,6 +535,74 @@ subroutine calc_resoln_function(h, tv, G, GV, US, CS)
...
+    do j=js,je ; do i=is,ie
+      Le(i,j) = MEKE%Le(i,j)
+      f(i,j) = max(0.25 * abs((G%CoriolisBu(I,J) + G%CoriolisBu(I-1,J-1)) + &
+                       (G%CoriolisBu(I-1,J) + G%CoriolisBu(I,J-1))), 1.0e-8 * US%s_to_T)
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90:581]: The hard-coded dimensional value of 1.0e-8 s-1 is not a particularly small number to use as a floor on the rotation.  What is the justification for this value, and could it be replaced by something else, such as a miniscule fraction of the planetary rotation rate omega?

File: src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90 [hunk 3]
```diff
@@ -1320,8 +1476,47 @@ subroutine VarMix_init(Time, G, GV, US, param_file, diag, CS)
                  units="m", default=-1.0, scale=GV%m_to_H)
     allocate(CS%ebt_struct(isd:ied,jsd:jed,GV%ke), source=0.0)
   endif
-  allocate(CS%BS_struct(isd:ied,jsd:jed,GV%ke), source=0.0)
-  CS%BS_struct(:,:,:) = 1.0
+
+
+  if (CS%BS_EBT_power>0. .and. CS%BS_use_sqg_struct) then
+    call MOM_error(FATAL, &
+      "calc_resoln_function: BS_EBT_POWER>0. &
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90:1483]: The indentation for a continuation line should be at least 4 spaces, according to the MOM6 style guide at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#white-space.

Comment (marshallward) [reply src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90]: ISO Fortran also requires line continuations (`&`) on both the end and beginning of split strings, e.g.
```
print *, "abcd &
    & efgh"
```

**Review by adcroft — APPROVED** [review]
Testing at https://gitlab.gfdl.noaa.gov/ogrp/mom6ci/MOM6/-/pipelines/25465

**Pull request merged**